### ISSUE TITLE
Support tracing logger?

### ISSUE THREAD

**Issue by WindSoilder** [dylanbstorey/pyo3-pylogger#5]
Hi, while reading example, I found that it uses `log` crate to initialize and log something.

I'm using tracing in application, and it seems that the log messages in python code doesn't show. 

I don't want to introduce another log crate to my application, is there a way to make it work with tracing?

**Comment by dylanbstorey** [comment]
> Hi, while reading example, I found that it uses `log` crate to initialize and log something.
> 
> I'm using tracing in application, and it seems that the log messages in python code doesn't show. 
> 
> I don't want to introduce another log crate to my application, is there a way to make it work with tracing?

Yes that should be doable,  im a bit busy atm but PRs are welcome.

**Comment by cpu** [comment]
FWIW I've been using this crate in an application that uses `tracing` through [tracing-log](https://docs.rs/tracing-log/latest/tracing_log/). Thank you! It's a very helpful crate :crab: :heart: :snake: 

Log messages from Python do appear in the expected place, with the correct level/message/source line number. The only foible in this setup I've observed is that the source file name is incorrect. PTAL at this branch for a fix that worked for me: https://github.com/dylanbstorey/pyo3-pylogger/pull/7

I'm not sure it's worth complicating this crate with `tracing` specific logic given the `tracing-log` crate exists and seems to work well out-of-box.

**Comment by LockedThread** [comment]
> FWIW I've been using this crate in an application that uses `tracing` through [tracing-log](https://docs.rs/tracing-log/latest/tracing_log/). Thank you! It's a very helpful crate 🦀 ❤️ 🐍
> 
> Log messages from Python do appear in the expected place, with the correct level/message/source line number. The only foible in this setup I've observed is that the source file name is incorrect. PTAL at this branch for a fix that worked for me: [#7](https://github.com/dylanbstorey/pyo3-pylogger/pull/7)
> 
> I'm not sure it's worth complicating this crate with `tracing` specific logic given the `tracing-log` crate exists and seems to work well out-of-box.

Are you able to share an example? I am trying to get this to work right now and am getting this: `Error: SetLoggerError(())`

**Comment by dylanbstorey** [comment]
@LockedThread - 

a quick search on cpu's activity last year yielded the following repo https://github.com/mudpuppy-rs/mudpuppy

Looks like its using this crate and tracing logging so should have examples/patterns that are reproducible.

**Comment by cpu** [comment]
> a quick search on cpu's activity last year yielded the following repo https://github.com/mudpuppy-rs/mudpuppy

Yup! That's the correct one.

I could probably pull out a smaller example if needed, but IIRC it's pretty straight-forward. The [Rust tracing config is here](https://github.com/mudpuppy-rs/mudpuppy/blob/ebbd2ef36a9cd0ccc96b55e8ac615a79faa3fac6/mudpuppy/src/config/logging.rs#L32-L82) and the [Python configuration is here](https://github.com/mudpuppy-rs/mudpuppy/blob/ebbd2ef36a9cd0ccc96b55e8ac615a79faa3fac6/mudpuppy/src/python.rs#L83-L97).

Hope that helps

**Comment by LockedThread** [comment]
> > a quick search on cpu's activity last year yielded the following repo https://github.com/mudpuppy-rs/mudpuppy
> 
> Yup! That's the correct one.
> 
> I could probably pull out a smaller example if needed, but IIRC it's pretty straight-forward. The [Rust tracing config is here](https://github.com/mudpuppy-rs/mudpuppy/blob/ebbd2ef36a9cd0ccc96b55e8ac615a79faa3fac6/mudpuppy/src/config/logging.rs#L32-L82) and the [Python configuration is here](https://github.com/mudpuppy-rs/mudpuppy/blob/ebbd2ef36a9cd0ccc96b55e8ac615a79faa3fac6/mudpuppy/src/python.rs#L83-L97).
> 
> Hope that helps

Thank you. I ended up doing pretty much the same thing, thank you for sending this though. Much appreciated.
